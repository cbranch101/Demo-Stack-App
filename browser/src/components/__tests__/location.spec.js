import React from "react"
import { mount } from "enzyme"
import { MemoryRouter } from "react-router"

import Location from "../location"

const PropContainer = props => {
    return (
        <div data-test="prop-container" data-props={props}>
            {JSON.stringify(props)}
        </div>
    )
}

const sel = id => `[data-test="${id}"]`

const getPropsFromMountedTestComponent = (props, location) => {
    const WrappedComponent = () => {
        return (
            <Location
                render={renderProps => {
                    return <PropContainer {...renderProps} />
                }}
            />
        )
    }
    const wrapper = mount(
        <MemoryRouter initialEntries={[location]} initialIndex={0} context={{}}>
            <WrappedComponent />
        </MemoryRouter>
    )
    return wrapper.find(sel("prop-container")).props()["data-props"]
}

test("should add the required props", () => {
    const locationToSet = {
        pathname: "/test",
        search: "?page=3"
    }
    const props = getPropsFromMountedTestComponent({}, locationToSet)
    const { location } = props
    const cleanedUpLocation = {
        ...location,
        key: undefined
    }
    expect({ ...props, location: cleanedUpLocation }).toMatchSnapshot()
})

test("should set a default of 1 if no page is provided", () => {
    const location = {
        pathname: "/test"
    }
    const { location: { query } } = getPropsFromMountedTestComponent({}, location)
    const expected = {
        page: 1
    }
    expect(query).toEqual(expected)
})

test("should set a default of 1 if no page is provided", () => {
    const location = {
        pathname: "/test"
    }
    const { location: { query } } = getPropsFromMountedTestComponent({}, location)
    const expected = {
        page: 1
    }
    expect(query).toEqual(expected)
})

test("should handle provided function functions", () => {
    const passedLocation = {
        pathname: "/test"
    }
    const { push } = getPropsFromMountedTestComponent({}, passedLocation)
    push(location => {
        expect(location.pathname).toEqual(passedLocation.pathname)
        return passedLocation
    })
})
