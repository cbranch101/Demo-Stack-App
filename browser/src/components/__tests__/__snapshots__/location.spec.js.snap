// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add the required props 1`] = `
Object {
  "location": Object {
    "hash": "",
    "key": undefined,
    "pathname": "/test",
    "query": Object {
      "page": 3,
    },
    "search": "?page=3",
  },
  "match": Object {
    "isExact": false,
    "params": Object {},
    "path": "/",
    "url": "/",
  },
  "push": [Function],
  "replace": [Function],
  "staticContext": undefined,
}
`;
